.github/workflows/ci-pypi-deploy.yml

```
@@ -0,0 +1,24 @@
+name: package-release
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+     - main
+  release:
+    types:
+      - published
+
+jobs:
+  build:
+    name: build and upload release to pypi
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
```

I have the usual comments ;)

```suggestion
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v5
```